File: comu/com.energyict.mdc.engine.monitor/src/main/web/js/mon/app/decorators/logging/DefaultEventDecorator.js
/*
 * Copyright (c) 2017 by Honeywell International Inc. All Rights Reserved
 */

Ext.define('CSMonitor.decorators.logging.DefaultEventDecorator', {
    alternateClassName: ['DefaultEventDecorator'],
    config: {
        event: undefined,
        occurrenceDateAsString: undefined
    },
    constructor: function(event) {
        this.setEvent(event);
        // the event's timestamp field contains the date as a String formatted by the backend:
        this.setOccurrenceDateAsString(this.getEvent().timestamp);
        return this;
    },
    asLogString: function() {
        var logString = this.occurrenceDateAsHTMLString() + ' - ' + this.getEvent()['class'] + ' {',
            attrib;
        for (attrib in this.getEvent()) {
            if (attrib !== 'timestamp' && attrib !== 'class') {
                logString += '\'' + attrib + '\': ' + this.getEvent()[attrib] + ', ';
            }
        }
        logString = logString.substring(0, logString.length - 2); // Subtract the last ', '
        logString += '}';
        return logString;
    },
    occurrenceDateAsHTMLString: function() {
        var htmlString = '<span class="logrecord-date">';
        htmlString += this.getOccurrenceDateAsString();
        htmlString += '</span>';
        return htmlString;
    }

});

File: comu/com.energyict.mdc.offline/src/main/java/com/energyict/mdc/engine/offline/gui/decorators/DefaultEventDecorator.java
package com.energyict.mdc.engine.offline.gui.decorators;

import java.util.HashMap;

/**
 * @author sva
 * @since 11/03/14 - 8:29
 */
public class DefaultEventDecorator implements EventDecorator {

    HashMap<String, Object> event;
    String occurenceDateAsString;

    public DefaultEventDecorator(HashMap<String, Object> event) {
        this.event = event;
        // the event's timestamp field contains the date as a String formatted by the backend:
        this.setOccurenceDateAsString((String) event.get("timestamp"));
    }

    public String asLogString() {
        String logString = getOccurenceDateAsString() + " - " + getEvent().get("clazz") + " {";

        for (String attrib : getEvent().keySet()) {
            if (!attrib.equals("timestamp") && !attrib.equals("clazz")) {
                logString += " " + attrib + ": " + this.getEvent().get(attrib);
            }
        }
        logString += "}";
        return logString;
    }

    public HashMap<String, Object> getEvent() {
        return event;
    }

    public void setEvent(HashMap<String, Object> event) {
        this.event = event;
    }

    public String getOccurenceDateAsString() {
        return occurenceDateAsString;
    }

    public void setOccurenceDateAsString(String occurenceDateAsString) {
        this.occurenceDateAsString = occurenceDateAsString;
    }
}
